py/mpstate.h

```diff
@@ -245,6 +250,14 @@ typedef struct _mp_state_thread_t {
     mp_obj_dict_t *dict_globals;
 
     nlr_buf_t *nlr_top;
+
+    #if MICROPY_PY_SYS_TRACE
+    mp_obj_t prof_trace_callback;
+    bool prof_callback_is_executing;
+    #endif // MICROPY_PY_SYS_TRACE
+    #if MICROPY_ACCESS_CODE_STATE
+    struct _mp_code_state_t *prof_code_state;
```

Comment:
It's unclear what was reworked, as one of the basic comments was that reference to the current code state being executed isn't really related to profiling per se, so shouldn't have "prof_" prefix.

Reply:
What was reworked is the tracing integration inside the *py/vm.c* and settrace callback invocation with proper frame and code objects. I renamed the pointer. It should be fixed now.